src/common/low_precision_transformations/src/fake_quantize_dequantization.cpp

```diff
@@ -135,20 +156,21 @@ bool FakeQuantizeDequantization::checkElementwise(const std::shared_ptr<ngraph::
 
     const size_t rank = partialShape.rank().get_length();
     if (constShape.size() == rank) {
-        if ((constShape[0] != 1ul) || (constShape[1] != channelsShapeVal)) {
+        if ((constShape[dimc] != channelsShapeVal)) {
             return false;
         }
-        for (size_t i = 2ul; i < constShape.size(); ++i) {
-            if (constShape[i] != 1ul) {
+        for (size_t i = 0ul; i < constShape.size(); ++i) {
+            if (constShape[i] != 1ul && i != dimc) {
                 return false;
             }
         }
     } else if (constShape.size() == (rank - 1)) {
-        if (constShape[0] != channelsShapeVal) {
+        dimc -= 1;
+        if (constShape[dimc] != channelsShapeVal) {
             return false;
         }
-        for (size_t i = 1ul; i < constShape.size(); ++i) {
-            if (constShape[i] != 1ul) {
+        for (size_t i = 0ul; i < constShape.size(); ++i) {
+            if (constShape[i] != 1ul && i != dimc) {
                 return false;
             }
         }
```

Comment:
Can we refactor this part to avoid code duplication? I suggest to correct `dimc` first and call all necessary checks after that.
Also, I would suggest to cover the case when difference between constant and data ranks more than 1 (using while loop). In this case we should pay attention whether we can shift `dimc`. (e.g. if dataRank = 4, constRank = 2 and dimc = 1, we can't do that and it's better to return false)

Reply:
Good idea, refactored